src/SFA.DAS.EmployerAccounts.UnitTests/Queries/GetEmployerAgreementQueryTests/GetEmployerAgreementTests.cs

```diff
@@ -85,8 +85,7 @@ public Task GetAgreementToSign_IfAgreementHasNotBeenSigned_ShouldAddCurrentUserA
                 assert: fixtures => Assert.AreEqual(user.FullName, fixtures.Response.EmployerAgreement.SignedByName));
         }
 
-        [Test]
```

Comment:
yeah..since the logic is moved to the EmployerController side, hence these tests are not required.

Reply:
I don't think the logic should be in the controller. What if someone calls the SignAgreement handler from a different execution path?

Reply:
The business rules need to be applied at the business layer.

Reply:
Any reason the SignAgreement handler can't just return an Unauthorized result if the user hasn't got permission to sign the agreement? This can then be used to redirect to the Error Page.